crates/autopilot/src/infra/blockchain/circuit_breaker.rs

```diff
@@ -0,0 +1,131 @@
+use {
+    crate::{
+        domain,
+        infra::blockchain::{
+            contracts::{deployment_address, Addresses},
+            Rpc,
+        },
+    },
+    ethcontract::transaction::TransactionResult,
+    primitive_types::H160,
+    std::{sync::Arc, time::Duration},
+    url::Url,
+};
+
+// Time between Ethereum blocks: 12 seconds
+// In order to fulfill Nyquist theory about sampling rate, we need to at least
+// sample twice the give frequency, in other words, half the period
+const SETTLEMENT_CHECK_SAMPLE_RATE: Duration = Duration::from_secs(6);
+
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub struct CircuitBreaker {
+    /// The authenticator contract used for allow-listing solvers to settle.
+    authenticator: contracts::GPv2AllowListAuthentication,
+    /// The safe module that is used to provide special role to EOA.
+    authenticator_role: contracts::Roles,
+    /// The EOA that is allowed to add/remove solvers.
+    authenticator_eoa: ethcontract::Account,
+    /// List of solvers to check
+    solvers: Vec<Url>,
```

Comment:
This would have to be a list of `H160` which are the onchain addresses of the respective solvers.